src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsPrisonOffenderEventListener.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.incidents
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.awspring.cloud.sqs.annotation.SqsListener
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.instrumentation.annotations.WithSpan
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.future.future
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.EventFeatureSwitch
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.SQSMessage
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.SynchronisationMessageType
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.INCIDENTS_SYNC_QUEUE_ID
+import java.util.concurrent.CompletableFuture
+
+@Service
+class IncidentsPrisonOffenderEventListener(
+  private val objectMapper: ObjectMapper,
+  private val eventFeatureSwitch: EventFeatureSwitch,
+) {
+
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @SqsListener(INCIDENTS_SYNC_QUEUE_ID, factory = "hmppsQueueContainerFactoryProxy")
+  @WithSpan(value = "syscon-devs-prisoner_from_nomis_incidents_queue", kind = SpanKind.SERVER)
+  fun onMessage(message: String): CompletableFuture<Void> {
+    log.debug("Received offender event message {}", message)
+    val sqsMessage: SQSMessage = objectMapper.readValue(message)
+    return asCompletableFuture {
+      when (sqsMessage.Type) {
+        "Notification" -> {
+          val eventType = sqsMessage.MessageAttributes!!.eventType.Value
+          if (eventFeatureSwitch.isEnabled(eventType)) {
+            when {
+              // INCIDENT-CHANGED-CASES, INCIDENT-CHANGED-PARTIES, INCIDENT-CHANGED-RESPONSES, INCIDENT-CHANGED-REQUIREMENTS,
+              eventType.startsWith("INCIDENT-CHANGED") ->
+                log.debug("Update event $eventType received", (sqsMessage.Message.fromJson()))
```

Comment:
this looks a strange log message, parameter passed but not used. Did you forget the `{}`?